saleor/order/migrations/0140_fix_order_undiscounted_total.py

```diff
@@ -0,0 +1,127 @@
+        # we should execute it only once after update `order` module.
+        if isinstance(sender, OrderAppConfig):
+            order_ids = list(kwargs.get("updated_orders_pks"))
+            send_order_updated.delay(order_ids)
+
+    Order = apps.get_model("order", "Order")
+    OrderLine = apps.get_model("order", "OrderLine")
+
+    # Take orders that has applied lines voucher discounts, but the discount is
+    # not visible in undiscounted price.
+    orders_to_update = Order.objects.filter(
+        Exists(
+            OrderLine.objects.filter(
+                order_id=OuterRef("id"), voucher_code__isnull=False
+            )
+        ),
+        total_gross_amount=F("undiscounted_total_gross_amount"),
+    )
+
+    updated_orders_pks = []
+    for batch_pks in queryset_in_batches(orders_to_update):
+        orders = Order.objects.filter(pk__in=batch_pks)
+        lines = OrderLine.objects.filter(order_id__in=orders.values("id")).values(
+            "order_id",
+            "undiscounted_total_price_gross_amount",
+            "total_price_gross_amount",
+            "undiscounted_total_price_net_amount",
+            "total_price_net_amount",
+        )
+        lines_discount_data = defaultdict(lambda: (0, 0))
+        for data in lines:
+            discount_amount_gross = (
+                data["undiscounted_total_price_gross_amount"]
+                - data["total_price_gross_amount"]
+            )
+            discount_amount_net = (
+                data["undiscounted_total_price_net_amount"]
+                - data["total_price_net_amount"]
+            )
+            current_discount_gross, current_discount_net = lines_discount_data[
+                data["order_id"]
+            ]
+            lines_discount_data[data["order_id"]] = (
+                current_discount_gross + discount_amount_gross,
+                current_discount_net + discount_amount_net,
+            )
+
+        for order in orders:
+            discount_amount_gross, discount_amount_net = lines_discount_data.get(
+                order.id
+            )
+            if discount_amount_gross > 0 or discount_amount_net > 0:
+                order.undiscounted_total_gross_amount += discount_amount_gross
+                order.undiscounted_total_net_amount += discount_amount_net
+
+                updated_orders_pks.append(order.id)
+
+        Order.objects.bulk_update(
+            orders,
+            [
+                "undiscounted_total_gross_amount",
+                "undiscounted_total_net_amount",
+            ],
+        )
+
+    # If we updated any order we should trigger `order_updated` after migrations
+    if updated_orders_pks:
+        updated_orders_pks = set(updated_orders_pks)
+        post_migrate.connect(
+            partial(on_migrations_complete, updated_orders_pks=updated_orders_pks),
+            weak=False,
+            dispatch_uid="send_order_updated",
+        )
```

Comment:
IMO we should fetch `sender`(`sender = registry.get_app_config("order")`) before `for` and:
```suggestion
        updated_orders_pks = set(updated_orders_pks)
        post_migrate.connect(
            partial(on_migrations_complete, updated_orders_pks=updated_orders_pks),
            weak=False,
            dispatch_uid="send_order_updated",
            sender=sender,
        )
```

Due to those changes we could drop `if isinstance(sender, OrderAppConfig):` from `on_migrations_complete`